Mathlib/Analysis/RCLike/Basic.lean

```diff
@@ -872,6 +872,20 @@ instance {A : Type*} [NonUnitalRing A] [StarRing A] [PartialOrder A] [StarOrdere
 
 scoped[ComplexOrder] attribute [instance] StarModule.instOrderedSMul
 
+theorem ofReal_mul_pos_iff (x : ℝ) (z : K) :
+    0 < x * z ↔ (x < 0 ∧ z < 0) ∨ (0 < x ∧ 0 < z) := by
+   simp_rw [pos_iff (z := x * z), pos_iff (z := z), neg_iff (z := z), re_ofReal_mul, im_ofReal_mul]
+   obtain hx | hx | hx := lt_trichotomy x 0
+   · simp_rw [mul_pos_iff, mul_eq_zero, hx, not_lt_of_gt hx, hx.ne, false_and, true_and,
+      false_or, or_false]
+   · simp_rw [hx, zero_mul, lt_self_iff_false, false_and, false_or]
+   · simp_rw [mul_pos_iff, mul_eq_zero, hx, not_lt_of_gt hx, hx.ne', false_and, true_and,
```

Comment:
does `simp only` work here? also, indenting is slightly nonstandard

Reply:
Yeah, sorry about the indenting. I'll fix that. 

`simp only` does work indeed. Do you think it's better than using `simp_rw`?

Reply:
I didn't know that. Thanks for the input. I was mostly using `simp_rw` because I like to see terms changing step by step. I find it quite rewarding :) 

BTW, David, if you have a little time can I interest you in having a look at #18166. I need the result way but I am not sure the way I prove it is the right one.

Reply:
Already on the case :-)